app/routes/EventDetail/EventDetailContainer.js

```diff
@@ -1,49 +1,56 @@
 import React, {Component} from 'react';
-import { AsyncStorage } from 'react-native';
+import { connect } from 'react-redux';
+import _ from 'lodash'
 
 import promises from '../../config/promises.js'
 import settings from '../../config/settings.js'
+import * as eventOperations from '../../operations/eventOperations'
 
 import EventDetail from './EventDetail.js';
 
 class EventDetailContainer extends Component {
-	constructor(props){
-		super();
-		this.state = {isPinned: props.pinned};
-	}
-
-	togglePinStatus(){
-		AsyncStorage.getItem(settings.keys.ID_TOKEN).then((idToken) => {
-			let url = (this.state.isPinned ? settings.urls.EVENTS_UNPIN_URL : settings.urls.EVENTS_PIN_URL) + this.props.id
-			promises.postWithToken(url, idToken)
-			.then((response) => {
-				this.setState({isPinned: !this.state.isPinned});
-			})
-			.catch((error) => {
-				console.error(error);
-			})
-		})
-	}
 
 	render() {
```

Comment:
ici tu pourrais faire un truc qui s'appelle [json object destructuring (ES6)](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Object_destructuring) pour être plus synthétique
```
const {idToken, id, pinned} = this.props
``` 
et donc, dans le context de ta fonction `render` tu peux acceder à ces variables sans avoir à mettre le `this.props` devant